addons/stock/models/stock_inventory.py

```diff
@@ -164,6 +170,48 @@ def action_reset_product_qty(self):
         return True
 
     def action_validate(self):
+        # Check the serial numbers uniquiness
+        serials = self.line_ids.filtered(lambda l: l.product_id.tracking == 'serial' and l.prod_lot_id and l.theoretical_qty < l.product_qty)
+        wiz_lines = []
+        for line in serials:
+            similar_quant = self.env['stock.quant'].search([
+                ('lot_id', '=', line.prod_lot_id.id),
+                ('location_id', '!=', line.location_id.id),
+                '|',
+                ('location_id.usage', 'not in', ('supplier', 'inventory', 'production')),
+                ('location_id.scrap_location', '=', True),
+                ('quantity', '>', 0)
+            ])
+            # Extract the quant to correct
+            for sq in similar_quant:
+                related_lines = self.line_ids.filtered(
+                    lambda line: line.correction_line == 'warning' and
```

Comment:
you probably should not check for explicit correction line created by the wizard. Manual correction should be okay also.